swap/src/bitcoin/wallet.rs

```diff
@@ -144,20 +211,36 @@ impl GetBlockHeight for Wallet {
 #[async_trait]
 impl TransactionBlockHeight for Wallet {
     async fn transaction_block_height(&self, txid: Txid) -> BlockHeight {
+        // todo: create this url using the join() api in the Url type
+        let url = format!("{}tx/{}/status", self.http_url, txid);
+        #[derive(Serialize, Deserialize, Debug, Clone)]
+        struct TransactionStatus {
+            block_height: Option<u32>,
+            confirmed: bool,
+        }
+        // todo: See if we can make this error handling more elegant
+        // errors
         #[derive(Debug)]
         enum Error {
```

Comment:
Sharing this error between the traits implementation would make it more readable.